src/storage/mod.rs

```diff
@@ -1847,47 +1847,32 @@ impl<E: Engine, L: LockManager, F: KvFormat> Storage<E, L, F> {
         }
     }
 
-    fn get_causal_ts(ts_provider: &Option<Arc<CausalTs>>) -> Result<Option<TimeStamp>> {
-        if let Some(p) = ts_provider {
-            match p.get_ts() {
-                Ok(ts) => Ok(Some(ts)),
-                Err(e) => Err(box_err!("Fail to get ts: {}", e)),
-            }
-        } else {
-            Ok(None)
-        }
-    }
+    async fn check_causal_ts_synced(ctx: &mut Context) -> Result<()> {
```

Comment:
I think it's no need to check for non-apiv2 scenario.

Reply:
Yesh, I added judgment, thanks.